feat(android): Add ANR docs around root cause analysis (#7180)
getsentry/sentry-docs: src/platforms/android/configuration/app-not-respond.mdx

</application>
```

### ApplicationExitInfo (v2)

This approach reads the [ApplicationExitInfo](https://developer.android.com/reference/android/app/ApplicationExitInfo) API on the next app launch
and asynchronously sends ANR events to Sentry for each ANR in the history, enriching only the latest one with breadcrumbs, contexts, tags, etc.

The integration reports ANR events with `mechanism:AppExitInfo`.

![ANR](app-not-respond.png)

## ANR Root Cause Analysis

> Sentry performs various root cause analyses to give you insights about why certain ANRs might appear. If a potential root cause is detected, it'll be displayed in a new section underneath the ANR stacktrace. Sentry is capable of detecting the following root causes:
> 

Reviewer: ```suggestion
Sentry performs various root cause analyses to give you insights about why certain ANRs might appear. If a potential root cause is detected, it'll be displayed in a new section below the ANR stacktrace. Sentry can detect the following root causes:

```

### Performance Issues

> If an ANR is connected to one of the Performance Issues detected by Sentry (for example, [File I/O on Main Thread](https://docs.sentry.io/product/issues/issue-details/performance-issues/file-main-thread-io/) or [DB on Main Thread](https://docs.sentry.io/product/issues/issue-details/performance-issues/db-main-thread-io/)), the offending span will be linked to the ANR event, showing you exactly which slow operation contributed to this ANR.
> 
> ![ANR Root Cause File I/O](anr-rootcause-fileio.png)

Reviewer: ```suggestion
If an ANR is connected to one of the Performance Issues detected by Sentry (for example, [File I/O on Main Thread](/product/issues/issue-details/performance-issues/file-main-thread-io/) or [DB on Main Thread](/product/issues/issue-details/performance-issues/db-main-thread-io/)), the offending span will be linked to the ANR event, showing you exactly which slow operation is to blame.

![ANR Root Cause File I/O](anr-rootcause-fileio.png)
```
We don't need to include the root folder of the url if we're redirecting people to our own docs :)

Author: yeah I know, I guess I was being lazy here, sorry 😬

### Common Patterns

> This detector looks at the stacktrace of the ANR and tries to identify common pitfalls, such as accessing files, assets or heavy resources on the main thread. If an offending stack frame is found, it will be highlighted in the stacktrace alongside some materials on how to prevent this type of ANR:
> 

Reviewer: ```suggestion
This detector looks at the stacktrace of an ANR and tries to identify common pitfalls, such as accessing files, assets, or heavy resources on the main thread. If an offending stack frame is found, it'll be highlighted in the stacktrace with some information about how to prevent this type of ANR:

```

![ANR Root Cause Suspect Frame](anr-rootcause-frame.png)

### Deadlock Detection

> Sentry will try to detect deadlocks and point out potential reasons. If, for example, the main thread is blocked waiting on a lock object held by another thread, the suspect frame that is waiting on the lock will be highlighted. Additionally, the stacktrace of the offending thread that is holding the lock will be shown, as well as the offending frame of that thread:
> 

Reviewer: ```suggestion
Sentry will try to detect deadlocks and point out potential causes. If, for example, the main thread is blocked by waiting on a lock object that's held by another thread, the suspect frame that's waiting on the lock will be highlighted. The stacktrace of the offending thread that's holding the lock, as well as the offending frame of that thread will also be shown:

```

The integration reports ANR events with `mechanism:AppExitInfo`.

![ANR](app-not-respond.png)

# ANR Root Cause Analysis

> Sentry performs various root cause analysis to give you tips about why some certain ANRs might appear. If we've
> detected a potential root cause, it'll be displayed in a new section underneath the ANR stacktrace. Sentry is
> capable of detecting the following root causes:

Reviewer 2: ```suggestion
Sentry performs various root cause analyses to give you insights about why certain ANRs might appear. If a potential root cause is detected, it'll be displayed in a new section underneath the ANR stacktrace. Sentry is capable of detecting the following root causes:
```

## Performance Issues

> If an ANR is connected to one of the Performance Issues detected by Sentry (for example, [File I/O on Main Thread](https://docs.sentry.io/product/issues/issue-details/performance-issues/file-main-thread-io/) or [DB on Main Thread](https://docs.sentry.io/product/issues/issue-details/performance-issues/db-main-thread-io/)), we'll link the offending span to the ANR event, showing you exactly which slow operation contributed to this ANR.

Reviewer 2: ```suggestion
If an ANR is connected to one of the Performance Issues detected by Sentry (for example, [File I/O on Main Thread](https://docs.sentry.io/product/issues/issue-details/performance-issues/file-main-thread-io/) or [DB on Main Thread](https://docs.sentry.io/product/issues/issue-details/performance-issues/db-main-thread-io/)), the offending span will be linked to the ANR event, showing you exactly which slow operation contributed to this ANR.
```

![ANR Root Cause File I/O](anr-rootcause-fileio.png)

## Common Patterns

> Our detector looks at the stacktrace of the ANR and tries to identify common pitfalls, such as accessing files, assets or heavy resources on the main thread. If we've found an offending stack frame, we will highlight it in the stacktrace and provide some materials on how to prevent this type of ANR:

Reviewer 2: ```suggestion
This detector looks at the stacktrace of the ANR and tries to identify common pitfalls, such as accessing files, assets or heavy resources on the main thread. If an offending stack frame is found, it will be highlighted it in the stacktrace alongside some materials on how to prevent this type of ANR:
```

![ANR Root Cause Suspect Frame](anr-rootcause-frame.png)

## Deadlock Detection

> We also detect deadlocks, for example, if the main thread is blocked waiting on a lock object held by another thread. In this case, we'll highlight
> the suspect frame that is waiting on the lock as well as show the stacktrace of the offending thread that is holding the lock. We'll also highlight the frame
> of the offending thread that is holding the lock:

Reviewer 2: ```suggestion
Sentry will try to detect deadlocks and point out potential reasons. If, for example, the main thread is blocked waiting on a lock object held by another thread, the suspect frame that is waiting on the lock will be highlighted. Additionally, the stacktrace of the offending thread that is holding the lock will be shown, as well as the frame of the offending thread that is holding the lock:
```
